tools/cxbe/Logo.cpp

```diff
@@ -0,0 +1,188 @@
+{
+    std::vector<uint08> bitmap;
+
+    for (size_t i = 0; i < raw.size(); i++) {
+        uint08 color = raw[i] >> 4;
+
+        uint32 len = 1;
+
+        while (i+1 < raw.size() && len < 1023 && color == raw[i+1] >> 4) {
+            len++;
+            i++;
+        }
+
+        LogoRLE rle;
+        if (len < 8) {
+            rle.m_Eight.bType1 = 1;
+            rle.m_Eight.Len = len;
+            rle.m_Eight.Data = color;
+            bitmap.push_back(rle.m_Bytes[0]);
+        } else {
+            rle.m_Sixteen.bType1 = 0;
+            rle.m_Sixteen.bType2 = 1;
+            rle.m_Sixteen.Len = len;
+            rle.m_Sixteen.Data = color;
+            bitmap.push_back(rle.m_Bytes[0]);
+            bitmap.push_back(rle.m_Bytes[1]);
+        }
+    }
+
+    return bitmap;
+}
+
+static void skipComments (std::ifstream &file)
+{
+    while (true) {
+        char c;
+
+        if (!file.get(c).good())
+            throw std::runtime_error("PGM file is corrupt");
+
+        if (c != '#') {
+            file.unget();
+            break;
+        }
+
+        while (true) {
+            file.get(c);
+            if (c == '\n' || c == '\r') {
+                break;
+            }
+        }
+
+    }
+}
+
+static uint32 readUintAscii (std::ifstream &file)
+{
+    std::stringstream s;
+
+    while (true) {
+        char c;
+
+        if (!file.get(c).good())
+            throw std::runtime_error("PGM file is corrupt");
+
+        if (c == ' ' || c == '\t' || c == '\n' || c == '\r') {
+            break;
+        } else if (c == '#') {
+            file.unget();
+            skipComments(file);
+            break;
+        } else if (c < '0' || c > '9') {
+            throw std::runtime_error("Invalid token in pgm file at offset " + std::to_string(file.tellg()));
+        }
+        s << c;
+    }
+
+    uint32 value;
+    s >> value;
+    return value;
+}
+
+std::vector<uint08> pgmToLogoBitmap (const char *filename)
+{
+    std::vector<uint08> raw;
+
+    std::ifstream file(filename, std::ios::binary);
+    if (!file.good())
+        throw std::runtime_error("Unable to open PGM file: " + std::string(filename));
+
+    char magic[3];
+    file.read(magic, 3);
+    if (magic[0] != 'P' || magic[1] != '5' ||
+            (magic[2] != ' ' && magic[2] != '\t' && magic[2] != '\n' && magic[2] != '\r'))
+        throw std::runtime_error("PGM has incorrect header. Only binary PGM files are supported.");
+
+    skipComments(file);
+    uint32 width = readUintAscii(file);
+    skipComments(file);
+    uint32 height = readUintAscii(file);
+    skipComments(file);
+    uint32 maxval = readUintAscii(file);
+
+    if (width != 100 || height != 17) {
+        throw std::runtime_error("PGM has wrong dimensions, 100x17 required\n");
+    }
+
+    if (maxval != 255) {
+        throw std::runtime_error("PGM has incompatible maxvalue, 255 required\n");
+    }
+
+    raw.resize(100*17);
+    file.read((char *)raw.data(), 100*17);
+    if (file.gcount() != 100*17) {
```

Comment:
Can we move the expression `100*17` to some constant as it's repeated multiple times here